plugins/outputs/cloudwatch/cloudwatch.go

```diff
@@ -161,9 +179,28 @@ func PartitionDatums(size int, datums []*cloudwatch.MetricDatum) [][]*cloudwatch
 	return partitions
 }
 
+// Make a MetricDatum from telegraf.Metric. It would check if all required fields of
+// cloudwatch.StatisticSet are available. If so, it would build MetricDatum from statistic values.
+// Otherwise, it would make MetricDatum from each field in a Point.
+func BuildMetricDatum(buildStatistic bool, point telegraf.Metric) []*cloudwatch.MetricDatum {
+
+	// If not enable, just take all metrics as value datums.
+	if !buildStatistic {
+		return BuildValueMetricDatum(point)
+	}
+
+	// Try to parse statisticSet first, then build statistic/value datum accordingly.
+	set, ok := getStatisticSet(point)
```

Comment:
If getStatisticSet() could not get all required fields, it just keeps the original behavior and sends them as independent metrics. It would generate 4 metrics to cloudwatch: cpu_foo_min, cpu_foo_max, cpu_bar_min and cpu_bar_max.

Reply:
Sorry, my example was somewhat lazy.  What if there are two full sets of statistics in a single `telegraf.Metric`, does it handle this case?
```
cpu foo_min=1,foo_max=2,foo_sum=4,foo_count=2,bar_min=1,bar_max=2,bar_sum=4,bar_count=2
```

Reply:
Good point. Will support this multiple sets case.